Source/loadsave.cpp

```diff
@@ -183,7 +183,7 @@ static void LoadItemData(ItemStruct *pItem)
 	tbuff += 1; // Alignment
 	CopyInt(tbuff, &pItem->_iStatFlag);
 	CopyInt(tbuff, &pItem->IDidx);
-	if (!gbIsHellfireSaveGame) {
+	if (!gbIsHellfireSaveGame && gbIsHellfire) {

```

Comment:
Mapping is to the ids in the static list and do not change depending if hellfire behavior is enabled or not.

```suggestion
	if (!gbIsHellfireSaveGame) {
```